packages/ui/src/theme/tokens/components/button.ts

```diff
@@ -1,4 +1,155 @@
-export const button = {
+import {
+  AlignItemsValue,
+  BackgroundColorValue,
+  BorderColorValue,
+  BorderRadiusValue,
+  BorderStyleValue,
+  BorderWidthValue,
+  BoxShadowValue,
+  ColorValue,
+  DesignToken,
+  FontSizeValue,
+  FontWeightValue,
+  JustifyContentValue,
+  LineHeightValue,
+  SpaceValue,
+  TransitionDurationValue,
+} from '../types/designToken';
+
+interface StateTokens {
+  color: DesignToken<ColorValue>;
+  backgroundColor: DesignToken<BackgroundColorValue>;
+  borderColor: DesignToken<BorderColorValue>;
+}
+
+interface StateWithShadowTokens extends StateTokens {
+  boxShadow: DesignToken<BoxShadowValue>;
+}
+
+interface PrimaryVariationTokens {
+  borderWidth: DesignToken<BorderWidthValue>;
+  borderStyle: DesignToken<BorderStyleValue>;
+  borderColor: DesignToken<BorderColorValue>;
+  backgroundColor: DesignToken<BackgroundColorValue>;
+  color: DesignToken<ColorValue>;
+  _disabled: {
```

Comment:
This could be `_disabled: StateTokens` right?

Reply:
Huh, I thought I did that. I got a bit lost in so many components 😅